src/algorithms/tracking/TrackerSourceLinker.cc

```diff
@@ -80,19 +80,24 @@ eicrecon::TrackerSourceLinkerResult *eicrecon::TrackerSourceLinker::produce(std:
 
         Acts::Vector2 loc = Acts::Vector2::Zero();
         Acts::Vector2 pos;
+        auto hit_det = hit->getCellID()&0xFF;
+        auto onSurfaceTolerance = 0.1*Acts::UnitConstants::um;      // By default, ACTS uses 0.1 micron as the on surface tolerance
+        if (hit_det==150){
+         onSurfaceTolerance = 1*Acts::UnitConstants::um;           // FIXME Ugly hack for testing B0. Should be a way to increase this tolerance in geometry.
+        }
```

Comment:
As discussed, I think this can be modified to use the `ActsGeometryProvider` instead of just the `dd4hep::rec::CellIDPositionConverter`. Then we can add a `dd4hep::rec::CellIDPositionConverter` hook in the `ActsGeometryProvider` but also use it to access the constants with something like
```suggestion
        if (hit_det==acts_geo->detector()->constant<int>("B0Tracker_Station_1_ID")){
         onSurfaceTolerance = 1*Acts::UnitConstants::um;           // FIXME Ugly hack for testing B0. Should be a way to increase this tolerance in geometry.
        }
```

Reply:
I was thinking of using the [dd4hep geometry service](https://github.com/eic/EICrecon/blob/158e3e0364a95965b249b0ffac16534e8123c6ad/src/services/geometry/dd4hep/JDD4hep_service.cc#L35) because the ActsGeometryProvider doesn't seem to provide a hook to retrieve the DD4hep geo. Other places seem to use JServiceLocator or GetApplication() to grab the DD4hep geometry service but i am having a hard time  getting it to work (https://github.com/eic/EICrecon/blob/158e3e0364a95965b249b0ffac16534e8123c6ad/src/global/tracking/CKFTracking_factory.cc#L30).

Reply:
Either way would be fine. I was just thinking `ActsGeometryProvider::dd4hepDetector()` ,https://github.com/eic/EICrecon/blob/main/src/algorithms/tracking/ActsGeometryProvider.h#L62 since that's already in theat class.

Reply:
Never mind. You are right. I was looking in acts geo service instead of the acts geometry provider class and confused myself. I have applied the fix and seems to work I think. This is 1000 events with brycecanyon.
![image](https://github.com/eic/EICrecon/assets/7409132/9950e753-c5ce-4cd4-9624-9d8abca39fd0)